### PR TITLE
fix: transaction sender method for type 0 and 1 transactions

### PR DESCRIPTION
**Pull request by vicentevieytes**
**Motivation**

Many of the ef_tests that should be trivial to pass are failing, tests with type 0 and type 1 transactions are failing because the sender of the transactions is being incorrectly derived from the transaction's message and signature.

**Description**

- Introduces the following function to correctly derive the chain_id from the "v" field in legacy transactions (T_w in the yellow paper)
```rust
    fn derive_legacy_chain_id(v: U256) -> Option<u64>
```
- Modifies the sender() method for legacy and eip2930 transactions to calculate the sender correctly.
- Modifies execute_test to run all transactions in every block from the fixture.
- Modifies execute_test to assert that execute_tx returns an Ok result value instead of expecting it to be an ExecutionResult::Success

- Gets the following test suites from ef_test to pass:
  - eip1153_tstore/
  - eip4788_beacon_root/
  - eip5656_mcopy/
  - eip7516_blobgasfee/

Partially closes https://github.com/lambdaclass/ethereum_rust/issues/151

### CODE REVIEW

File: cmd/ef_tests/test_runner.rs [hunk 1]
```diff
@@ -4,40 +4,43 @@ use crate::types::{Account, TestUnit};
...
+                &test
+                    .blocks
+                    .first()
+                    .as_ref()
+                    .unwrap()
+                    .block_header
+                    .clone()
+                    .unwrap()
+                    .into(),
+                &mut build_evm_state_from_prestate(&test.pre),
```
Comment (mpaulucci) [inline cmd/ef_tests/test_runner.rs:36]: here you are using the same state for each transaction in a block. The n+1 transaction should use the state generated after executing the n transaction. This https://github.com/lambdaclass/ethereum_rust/pull/206 should fix it

File: crates/core/types/transaction.rs [hunk 2]
```diff
@@ -539,7 +550,7 @@ impl Transaction {
 
     pub fn chain_id(&self) -> Option<u64> {
```
Comment (mpaulucci) [inline crates/core/types/transaction.rs:551]: why does this return an Option if all the values are `Some`?

File: cmd/ef_tests/test_runner.rs (hunk 1 above)
Comment (vicentevieytes) [reply cmd/ef_tests/test_runner.rs]: By moving the evm_state out of the loop this is fixed, now the eip6780_selfdestruct tests which had the "nonce too high" issue are passing.

**Review by mpaulucci — APPROVED** [review]
nice work!

**Pull request merged**